src/components/Wallet/WalletSpend.jsx

```diff
@@ -31,41 +32,43 @@ import {
   SPEND_STEP_PREVIEW,
   SPEND_STEP_SIGN,
   setSpendStep as setSpendStepAction,
+  deleteChangeOutput as deleteChangeOutputAction,
 } from "../../actions/transactionActions";
+import { naiveCoinSelection } from "../../utils";
 
 // Components
 import NodeSet from "./NodeSet";
 import OutputsForm from "../ScriptExplorer/OutputsForm";
 import WalletSign from "./WalletSign";
 import TransactionPreview from "./TransactionPreview";
+import { bigNumberPropTypes } from "../../proptypes/utils";
 
 class WalletSpend extends React.Component {
   outputsAmount = new BigNumber(0);
 
+  coinSelection = naiveCoinSelection;
+
   feeAmount = new BigNumber(0);
 
-  componentDidUpdate() {
-    const { autoSpend, finalizedOutputs } = this.props;
-    if (autoSpend && !finalizedOutputs) {
-      setTimeout(this.selectCoins, 0);
+  componentDidUpdate = (prevProps) => {
+    const { finalizedOutputs } = this.props;
+    if (finalizedOutputs && !prevProps.finalizedOutputs) {
+      this.showPreview();
     }
-  }
-
-  componentDidMount = () => {
-    const { changeNode, setChangeAddress, autoSpend } = this.props;
-    if (autoSpend) setChangeAddress(changeNode.multisig.address);
```

Comment:
only determined once the outputs are set, not on mount anymore. This allows for a spend all tx and for a tx that might not need change.